simplecpp.cpp

```diff
@@ -131,7 +131,7 @@ const std::string simplecpp::Location::emptyFileName;
 
 void simplecpp::Location::adjust(const std::string &str)
 {
-    if (str.find_first_of("\r\n") == std::string::npos) {
+    if (!strpbrk(str.c_str(), "\r\n")) {
```

Comment:
I find it strange that this would be an optimisation. Is strpbrk optimised better?

However I think it obfuscates the code. I do not intuitively know how strpbrk works.

Reply:
callgrind showed and there's this
https://www.codeproject.com/Articles/18473/find-first-of-A-performance-pitfall-among-the-STL#A30. I used the latest GCC so it seems this is still true although that article is 12 years old.

`strpbrk` is the same as `find_first_of`.

Reply:
Running `simplecpp` on `-Ilib lib/valueflow.cpp`:
GCC `335,129,681` -> `325,111,787`
Clang `323,433,151` -> `313,133,608`